core/src/test/java/io/grpc/MetadataTest.java

```diff
@@ -145,6 +145,19 @@ public void roundTripIntegerMarshaller() {
     roundTripInteger(0x87654321);
   }
 
+  @Test
+  public void timeoutTest() {
```

Comment:
Since you added the parser, it is probably a good idea to test it at this time too. Otherwise I think we will start using it and not realize it doesn't have tests. Should be easy.